pyrad/tests/testPacket.py

```diff
@@ -186,6 +231,42 @@ def testVerifyReply(self):
         self.assertEqual(self.packet.VerifyReply(reply), False)
         reply.authenticator = self.packet.authenticator
 
+    def testVerifyReplyDuplicateAttributes(self):
+        reply = self._create_reply_with_duplicate_attributes(self.packet)
+        self.assertTrue(self.packet.VerifyReply(
+            reply=reply,
+            rawreply=reply.raw_packet))
+
+    def testVerifyMessageAuthenticator(self):
+        reply = self.packet.CreateReply(**{
+            'Test-String': 'test',
+            'Test-Integer': 3,
+        })
+        reply.code = packet.AccessAccept
+        reply.add_message_authenticator()
+        reply._refresh_message_authenticator()
+        self.assertTrue(reply.verify_message_authenticator(
+            secret=six.b('secret'),
+            original_authenticator=self.packet.authenticator,
+            original_code=self.packet.code))
+
+        self.assertFalse(reply.verify_message_authenticator(
+            secret=six.b('bad_secret'),
+            original_authenticator=self.packet.authenticator,
+            original_code=self.packet.code))
+
+        self.assertFalse(reply.verify_message_authenticator(
+            secret=six.b('secret'),
+            original_authenticator=six.b('bad_authenticator'),
+            original_code=self.packet.code))
+
+    def testVerifyMessageAuthenticatorDuplicateAttributes(self):
+        reply = self._create_reply_with_duplicate_attributes(self.packet)
+        self.assertTrue(reply.verify_message_authenticator(
+            secret=six.b('secret'),
+            original_authenticator=self.packet.authenticator,
+            original_code=packet.AccessRequest))
```

Comment:
The test covering the bug fix. This test fails without the changes made to packet.py in this commit.